From eed5b8fe481b39f3d4b69a6398a7d48a19b859a4 Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Wed, 19 Oct 2011 23:47:44 +0200
Subject: [PATCH 09/19] migration: don't "write" when migration is not active

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <aeb751c7c2a35323d492b8691b863311bb7ef05d.1319066771.git.quintela@redhat.com>
Patchwork-id: 34435
O-Subject: [PATCH qemu-kvm RHEL-6.2 09/16] migration: don't "write" when migration is not active
Bugzilla: 669581
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

If migration is not active, just ignore writes.

Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 migration.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 migration.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/migration.c b/migration.c
index ff60b3b..dd65bf7 100644
--- a/migration.c
+++ b/migration.c
@@ -340,6 +340,10 @@ ssize_t migrate_fd_put_buffer(void *opaque, const void *data, size_t size)
     FdMigrationState *s = opaque;
     ssize_t ret;
 
+    if (s->state != MIG_STATE_ACTIVE) {
+        return -EIO;
+    }
+
     do {
         ret = s->write(s, data, size);
     } while (ret == -1 && ((s->get_error(s)) == EINTR));
-- 
1.7.4.4

